common/functions/src/scalars/semi_structureds/get.rs

```diff
@@ -0,0 +1,249 @@
+            return Err(ErrorCode::SyntaxException(
+                "Bad compound object's field path name: '' in GET_PATH",
+            ));
+        }
+        let definition = std::str::from_utf8(v).unwrap();
+        let mut tokenizer = Tokenizer::new(dialect, definition);
+        match tokenizer.tokenize() {
+            Ok(tokens) => match Parser::new(tokens, dialect).parse_map_keys() {
+                Ok(values) => {
+                    let path_key: Vec<DataValue> = values
+                        .iter()
+                        .map(|v| match v {
+                            Value::Number(value, _) => {
+                                DataValue::try_from_literal(value, None).unwrap()
+                            }
+                            Value::SingleQuotedString(value) => {
+                                DataValue::String(value.clone().into_bytes())
+                            }
+                            Value::ColonString(value) => {
+                                DataValue::String(value.clone().into_bytes())
+                            }
+                            Value::PeriodString(value) => {
+                                DataValue::String(value.clone().into_bytes())
+                            }
+                            _ => DataValue::Null,
+                        })
+                        .collect();
+
+                    path_keys.push(path_key);
+                }
+                Err(parse_error) => return Err(ErrorCode::from(parse_error)),
+            },
+            Err(tokenize_error) => {
+                return Err(ErrorCode::SyntaxException(format!(
+                    "Can not tokenize definition: {}, Error: {:?}",
+                    definition, tokenize_error
+                )))
+            }
+        }
+    }
+    Ok(path_keys)
+}
+
+fn build_path_keys(column: &ColumnRef) -> Result<Vec<Vec<DataValue>>> {
+    if column.is_const() {
+        let const_column: &ConstColumn = Series::check_get(column)?;
+        return build_path_keys(const_column.inner());
+    }
+
+    let mut path_keys: Vec<Vec<DataValue>> = vec![];
+    for i in 0..column.len() {
+        path_keys.push(vec![column.get(i)]);
+    }
+    Ok(path_keys)
+}
+
+fn extract_value_by_path(
+    column: &ColumnRef,
+    path_keys: Vec<Vec<DataValue>>,
+    input_rows: usize,
+    ignore_case: bool,
+) -> Result<ColumnRef> {
+    let column: &JsonColumn = if column.is_const() {
+        let const_column: &ConstColumn = Series::check_get(column)?;
+        Series::check_get(const_column.inner())?
+    } else {
+        Series::check_get(column)?
+    };
+
+    let mut builder = NullableColumnBuilder::<JsonValue>::with_capacity(input_rows);
+    for path_key in path_keys.iter() {
+        if path_key.is_empty() {
+            for _ in 0..column.len() {
+                builder.append_null();
+            }
+            continue;
+        }
+        for v in column.iter() {
+            let mut found_value = true;
```

Comment:
Will found_value defaults to false make this more simple?

Reply:
ok